### ISSUE TITLE
AttributeError: socket_method when invoking 'zato.outgoing.zmq.create' service

### ISSUE THREAD

**Issue by rafal-k** [zatosource/zato#731]
Attempts to create a zmq outgoing connection result in AttributeError.

Contents of a server log:

```
2017-01-10 14:45:19,800 - ERROR - 18834:DummyThread-42 - zato.outgoing.zmq.create:80 - Could not create an outgoing ZeroMQ connection, e:[Traceback (most recent call last):
  File "/home/rafal/Projects/Zato/zato/code/zato-server/src/zato/server/service/internal/outgoing/zmq.py", line 66, in handle
    item.socket_method = input.socket_method
  File "/home/rafal/Projects/Zato/zato/code/eggs/bunch-1.0.1-py2.7.egg/bunch/__init__.py", line 121, in __getattr__
    raise AttributeError(k)
AttributeError: socket_method
]
2017-01-10 14:45:19,800 - WARNING - 18834:DummyThread-42 - zato.server.service:394 - Traceback (most recent call last):
  File "/home/rafal/Projects/Zato/zato/code/zato-server/src/zato/server/service/__init__.py", line 382, in update_handle
    self._invoke(service, channel)
  File "/home/rafal/Projects/Zato/zato/code/zato-server/src/zato/server/service/__init__.py", line 316, in _invoke
    service.handle()
  File "/home/rafal/Projects/Zato/zato/code/zato-server/src/zato/server/service/internal/outgoing/zmq.py", line 66, in handle
    item.socket_method = input.socket_method
  File "/home/rafal/Projects/Zato/zato/code/eggs/bunch-1.0.1-py2.7.egg/bunch/__init__.py", line 121, in __getattr__
    raise AttributeError(k)
AttributeError: socket_method

2017-01-10 14:45:19,800 - WARNING - 18834:DummyThread-42 - zato.server.service:409 - Exception in service `zato.outgoing.zmq.create`, e:`Traceback (most recent call last):
  File "/home/rafal/Projects/Zato/zato/code/zato-server/src/zato/server/service/__init__.py", line 398, in update_handle
    response = set_response_func(service, data_format=data_format, transport=transport, **kwargs)
  File "/home/rafal/Projects/Zato/zato/code/zato-server/src/zato/server/connection/http_soap/channel.py", line 292, in _set_response_data
    self.set_payload(service.response, data_format, transport, service)
  File "/home/rafal/Projects/Zato/zato/code/zato-server/src/zato/server/connection/http_soap/channel.py", line 383, in set_payload
    payload = response.payload.getvalue(False)
  File "/home/rafal/Projects/Zato/zato/code/zato-server/src/zato/server/service/reqresp/__init__.py", line 346, in getvalue
    elem_value = self._getvalue(name, item, is_sa_namedtuple, is_required, leave_as_is)
  File "/home/rafal/Projects/Zato/zato/code/zato-server/src/zato/server/service/reqresp/__init__.py", line 292, in _getvalue
    raise ZatoException(self.zato_cid, msg)
ZatoException: Expected elem:[id] not found in item:[{u'id': u'', u'name': u''}]
`
2017-01-10 14:45:19,801 - ERROR - 18834:DummyThread-42 - zato.server.connection.http_soap.channel:252 - Caught an exception, cid:`38c28304fe7a937cbef01c6c`, status_code:`500`, _format_exc:`Traceback (most recent call last):
  File "/home/rafal/Projects/Zato/zato/code/zato-server/src/zato/server/connection/http_soap/channel.py", line 206, in dispatch
    payload, worker_store, self.simple_io_config, post_data)
  File "/home/rafal/Projects/Zato/zato/code/zato-server/src/zato/server/connection/http_soap/channel.py", line 351, in handle
    params_priority=channel_item.params_pri)
  File "/home/rafal/Projects/Zato/zato/code/zato-server/src/zato/server/service/__init__.py", line 412, in update_handle
    raise Exception(exc_formatted)
Exception: Traceback (most recent call last):
  File "/home/rafal/Projects/Zato/zato/code/zato-server/src/zato/server/service/__init__.py", line 382, in update_handle
    self._invoke(service, channel)
  File "/home/rafal/Projects/Zato/zato/code/zato-server/src/zato/server/service/__init__.py", line 316, in _invoke
    service.handle()
  File "/home/rafal/Projects/Zato/zato/code/zato-server/src/zato/server/service/internal/outgoing/zmq.py", line 66, in handle
    item.socket_method = input.socket_method
  File "/home/rafal/Projects/Zato/zato/code/eggs/bunch-1.0.1-py2.7.egg/bunch/__init__.py", line 121, in __getattr__
    raise AttributeError(k)
AttributeError: socket_method
`
```

Request via curl:

```
curl $ZATO_API_TEST_SERVER/zato/json/zato.outgoing.zmq.create -u pubapi:pubapi -d '{"cluster_id": 1, "name": "lhdfaosidygpoylkhh", "is_active": "true", "address": "tcp://localhost:5555", "socket_type": "PUSH"}' -D -
HTTP/1.1 500 Internal Server Error
Server: Zato
Date: Tue, 10 Jan 2017 13:41:43 GMT
Connection: close
Transfer-Encoding: chunked
X-Zato-CID: f416d997a2f359e104d4226c

{"zato_env": {"details": "Traceback (most recent call last):\n  File \"/home/rafal/Projects/Zato/zato/code/zato-server/src/zato/server/connection/http_soap/channel.py\", line 206, in dispatch\n    payload, worker_store, self.simple_io_config, post_data)\n  File \"/home/rafal/Projects/Zato/zato/code/zato-server/src/zato/server/connection/http_soap/channel.py\", line 351, in handle\n    params_priority=channel_item.params_pri)\n  File \"/home/rafal/Projects/Zato/zato/code/zato-server/src/zato/server/service/__init__.py\", line 412, in update_handle\n    raise Exception(exc_formatted)\nException: Traceback (most recent call last):\n  File \"/home/rafal/Projects/Zato/zato/code/zato-server/src/zato/server/service/__init__.py\", line 382, in update_handle\n    self._invoke(service, channel)\n  File \"/home/rafal/Projects/Zato/zato/code/zato-server/src/zato/server/service/__init__.py\", line 316, in _invoke\n    service.handle()\n  File \"/home/rafal/Projects/Zato/zato/code/zato-server/src/zato/server/service/internal/outgoing/zmq.py\", line 66, in handle\n    item.socket_method = input.socket_method\n  File \"/home/rafal/Projects/Zato/zato/code/eggs/bunch-1.0.1-py2.7.egg/bunch/__init__.py\", line 121, in __getattr__\n    raise AttributeError(k)\nAttributeError: socket_method\n\n", "result": "ZATO_ERROR", "cid": "f416d997a2f359e104d4226c"}}
```

Zato version:  `Zato 3.0.0rc1+rev.fb634a91`

**Comment by rafal-k** [comment]
Creating a zmq outgoing connecting via web-admin doesn't work as well:

```
Traceback (most recent call last):
  File "/home/rafal/Projects/Zato/zato/code/zato-web-admin/src/zato/admin/web/views/__init__.py", line 426, in __call__
    response = self.req.zato.client.invoke(self.service_name, self.input_dict)
  File "/home/rafal/Projects/Zato/zato/code/zato-web-admin/src/zato/admin/middleware.py", line 78, in invoke
    raise Exception('CID: {}\nDetails: {}'.format(zato_env.get('cid'), zato_env.get('details')))
Exception: CID: a0b9c70b23b55fd04cfe1f99
Details: Traceback (most recent call last):
  File "/home/rafal/Projects/Zato/zato/code/zato-server/src/zato/server/connection/http_soap/channel.py", line 206, in dispatch
    payload, worker_store, self.simple_io_config, post_data)
  File "/home/rafal/Projects/Zato/zato/code/zato-server/src/zato/server/connection/http_soap/channel.py", line 351, in handle
    params_priority=channel_item.params_pri)
  File "/home/rafal/Projects/Zato/zato/code/zato-server/src/zato/server/service/__init__.py", line 382, in update_handle
    self._invoke(service, channel)
  File "/home/rafal/Projects/Zato/zato/code/zato-server/src/zato/server/service/__init__.py", line 316, in _invoke
    service.handle()
  File "/home/rafal/Projects/Zato/zato/code/zato-server/src/zato/server/service/internal/service.py", line 306, in handle
    response = func(id_, payload, channel, data_format, transport, serialize=True)
  File "/home/rafal/Projects/Zato/zato/code/zato-server/src/zato/server/service/__init__.py", line 484, in invoke
    return self.invoke_by_impl_name(self.server.service_store.name_to_impl_name[name], *args, **kwargs)
  File "/home/rafal/Projects/Zato/zato/code/zato-server/src/zato/server/service/__init__.py", line 467, in invoke_by_impl_name
    return self.update_handle(*invoke_args, **kwargs)
  File "/home/rafal/Projects/Zato/zato/code/zato-server/src/zato/server/service/__init__.py", line 412, in update_handle
    raise Exception(exc_formatted)
Exception: Traceback (most recent call last):
  File "/home/rafal/Projects/Zato/zato/code/zato-server/src/zato/server/service/__init__.py", line 382, in update_handle
    self._invoke(service, channel)
  File "/home/rafal/Projects/Zato/zato/code/zato-server/src/zato/server/service/__init__.py", line 316, in _invoke
    service.handle()
  File "/home/rafal/Projects/Zato/zato/code/zato-server/src/zato/server/service/internal/outgoing/zmq.py", line 66, in handle
    item.socket_method = input.socket_method
  File "/home/rafal/Projects/Zato/zato/code/eggs/bunch-1.0.1-py2.7.egg/bunch/__init__.py", line 121, in __getattr__
    raise AttributeError(k)
AttributeError: socket_method
```

**Comment by dsuch** [comment]
Fixed in main since 8ffdceefe954f2d51b66327c79edec349d5739e0.
